WinUIB2CApp/MSAL/MSALClientHelper.cs

```diff
@@ -0,0 +1,248 @@
+                .WithIosKeychainSecurityGroup("com.microsoft.adalcache");
+        }
+
+        /// <summary>
+        /// Initializes the public client application of MSAL.NET with the required information to correctly authenticate the user.
+        /// </summary>
+        /// <returns></returns>
+        public async Task<IAccount> InitializePublicClientAppAsync()
+        {
+            // Initialize the MSAL library by building a public client application
+            this.PublicClientApplication = this.PublicClientApplicationBuilder.Build();
+
+            await AttachTokenCache();
+            return await FetchSignedInUserFromCache().ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Initializes the public client application of MSAL.NET with the required information to correctly authenticate the user.
+        /// </summary>
+        /// <returns>An IAccount of an already signed-in user (if available)</returns>
+        public async Task<IAccount> InitializePublicClientB2CAppAsync(IntPtr? handle)
+        {
+            // Initialize the MSAL library by building a public client application for authenticating using WAM
+            this.PublicClientApplication = this.PublicClientApplicationBuilder
+                    .Build();
+
+            await AttachTokenCache();
+            return await FetchSignedInUserFromCache().ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Attaches the token cache to the Public Client app.
+        /// </summary>
+        /// <returns>IAccount list of already signed-in users (if available)</returns>
+        private async Task<IEnumerable<IAccount>> AttachTokenCache()
+        {
+            // Cache configuration and hook-up to public application. Refer to https://github.com/AzureAD/microsoft-authentication-extensions-for-dotnet/wiki/Cross-platform-Token-Cache#configuring-the-token-cache
+            var storageProperties = new StorageCreationPropertiesBuilder(AzureADB2CConfig.CacheFileName, AzureADB2CConfig.CacheDir).Build();
+            var msalcachehelper = await MsalCacheHelper.CreateAsync(storageProperties);
+            msalcachehelper.RegisterCache(this.PublicClientApplication.UserTokenCache);
+
+            // If the cache file is being reused, we'd find some already-signed-in accounts
+            return await this.PublicClientApplication.GetAccountsAsync().ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Signs in the user and obtains an Access token for a provided set of scopes
+        /// </summary>
+        /// <param name="scopes"></param>
+        /// <returns> Access Token</returns>
+        public async Task<string> SignInUserAndAcquireAccessToken(string[] scopes)
+        {
+            var existingUser = await FetchSignedInUserFromCache().ConfigureAwait(false);
+
+            try
+            {
+                // 1. Try to sign-in the previously signed-in account
+                if (existingUser != null)
```

Comment:
You do not need to check this. AcquireTokenSilent(null) will throw `MsalUiRequiredException`